include/usearch/index_dense.hpp

```diff
@@ -875,46 +843,16 @@ class index_dense_gt {
                                             serialization_config_t config = {}, //
                                             progress_at&& progress = {}) {
 
-        // Discard all previous memory allocations of `vectors_tape_allocator_`
+        // Discard all previous memory allocations of
         reset();
 
         // Infer the new index size
         serialization_result_t result;
-        std::uint64_t matrix_rows = 0;
-        std::uint64_t matrix_cols = 0;
-
-        // We may not want to load the vectors from the same file, or allow attaching them afterwards
-        if (!config.exclude_vectors) {
-            // Save the matrix size
-            if (!config.use_64_bit_dimensions) {
-                std::uint32_t dimensions[2];
-                if (!input(&dimensions, sizeof(dimensions)))
-                    return result.failed("Failed to read 32-bit dimensions of the matrix");
-                matrix_rows = dimensions[0];
-                matrix_cols = dimensions[1];
-            } else {
-                std::uint64_t dimensions[2];
-                if (!input(&dimensions, sizeof(dimensions)))
-                    return result.failed("Failed to read 64-bit dimensions of the matrix");
-                matrix_rows = dimensions[0];
-                matrix_cols = dimensions[1];
-            }
-            // Load the vectors one after another
-            vectors_lookup_.resize(matrix_rows);
-            for (std::uint64_t slot = 0; slot != matrix_rows; ++slot) {
-                byte_t* vector = vectors_tape_allocator_.allocate(matrix_cols);
-                if (!input(vector, matrix_cols))
-                    return result.failed("Failed to read vectors");
-                vectors_lookup_[slot] = vector;
-            }
-        }
+        index_dense_head_buffer_t buffer;
+        storage_.load_vectors_from_stream(input, buffer, config);
 
         // Load metadata and choose the right metric
         {
-            index_dense_head_buffer_t buffer;
-            if (!input(buffer, sizeof(buffer)))
```

Comment:
`storage_at::load_vectors_from_stream` takes a generic buffer and reads off bytes into it from the specified section in the storage buffer, per storage format spec